v0/destinations/google_cloud_function/util.js

```diff
@@ -0,0 +1,97 @@
+const { defaultBatchRequestConfig, CustomError } = require("../../util");
+
+const { TRIGGERTYPE } = require("./config");
+
+const validateDestinationConfig = ({ Config }) => {
+  if (!Config.googleCloudFunctionUrl) {
+    throw new CustomError("[GCF]:: Url not found. Aborting", 400);
+  }
+  if (Config.TriggerType === "https") {
+    // Auth Mandatory
+    if (!Config.gcloudAuthorization) {
+      throw new CustomError("[GCF]:: Access Token not found. Aborting", 400);
+    }
+  }
+};
+
+function addHeader(response, Config) {
+  const { TriggerType, apiKeyId, gcloudAuthorization } = Config;
+  let basicAuth;
+  if (apiKeyId) {
+    basicAuth = Buffer.from(`apiKey:${apiKeyId}`).toString("base64");
+  }
+
+  response.headers = { "content-type": "application/json" };
+  if (TRIGGERTYPE.HTTPS === TriggerType.toLowerCase()) {
+    response.headers.Authorization = `bearer ${gcloudAuthorization}`;
+  }
+  if (basicAuth) {
+    if (TRIGGERTYPE.HTTPS === TriggerType.toLowerCase()) {
+      response.headers.ApiKey = `Basic ${basicAuth}`;
+    } else {
+      response.headers.Authorization = `Basic ${basicAuth}`;
+    }
+  }
+}
+
+function generateBatchedPayloadForArray(events) {
```

Comment:
Both `generateBatchedPayloadForArray` and `generateBatchedPayload` functions are almost same. Except very few changes. Please avoid redundant code .